src/hera/workflows/workflow_template.py

```diff
@@ -153,5 +163,21 @@ def build(self) -> TWorkflow:
             ),
         )
 
+    def create_as_workflow(self, wait: bool = False, poll_interval: int = 5) -> Workflow:
+        """Run this WorkflowTemplate instantly as a Workflow.
+
+        Note: this does not require the WorkflowTemplate to exist on the cluster
+        """
+
+        workflow = self.build()
+        workflow.kind = "Workflow"
+        if workflow.metadata.name:
+            # In case of a name being > TRUNCATE_LENGTH, we need to truncate the value to assign to generate_name
+            # otherwise the random suffix will make the generated name too long
+            workflow.metadata.generate_name = workflow.metadata.name[:TRUNCATE_LENGTH] + "-"
+            workflow.metadata.name = None
```

Comment:
This is an implicit modification of a user submitted workflow that Hera will perform - what's a good way to "advertise" this is happening? Perhaps a log? 🤔 Also, should this logic be isolated and shared in a validator, perhaps? Maybe wf template + cluster wf template might also benefit from inheriting this. Thoughts?